src/angular-websocket.js

```diff
@@ -1,398 +1,392 @@
-(function() {
-  'use strict';
-
-  var noop = angular.noop;
-  var objectFreeze  = (Object.freeze) ? Object.freeze : noop;
-  var objectDefineProperty = Object.defineProperty;
-  var isString   = angular.isString;
-  var isFunction = angular.isFunction;
-  var isDefined  = angular.isDefined;
-  var isObject   = angular.isObject;
-  var isArray    = angular.isArray;
-  var forEach    = angular.forEach;
-  var arraySlice = Array.prototype.slice;
-  // ie8 wat
-  if (!Array.prototype.indexOf) {
-    Array.prototype.indexOf = function(elt /*, from*/) {
-      var len = this.length >>> 0;
-      var from = Number(arguments[1]) || 0;
-      from = (from < 0) ? Math.ceil(from) : Math.floor(from);
-      if (from < 0) {
-        from += len;
-      }
-
-      for (; from < len; from++) {
-        if (from in this && this[from] === elt) { return from; }
-      }
-      return -1;
-    };
-  }
-
-  // $WebSocketProvider.$inject = ['$rootScope', '$q', '$timeout', '$websocketBackend'];
-  function $WebSocketProvider($rootScope, $q, $timeout, $websocketBackend) {
+import angular from 'angular';
+import * as ws from 'ws';
+
+var Socket = (ws.Client || ws.client || ws);
```

Comment:
This check previously existed inside of a function call. It is preferable to do these kinds of checks outside of any particular function. Also, the [try/catch](https://github.com/AngularClass/angular-websocket/blob/master/src/angular-websocket.js#L365-L368) block that silently swallowed errors was a bit of a code smell to me. UMD fixes the problem (I think) you were trying to solve via that mechanism.